libbeat/template/load.go

```diff
@@ -124,6 +124,30 @@ func (l *ESLoader) Load(config TemplateConfig, info beat.Info, fields []byte, mi
 		return fmt.Errorf("failed to load template: %w", err)
 	}
 	l.log.Infof("Template with name %q loaded.", templateName)
+
+	// if JSON template is loaded and it is not a data stream
+	// we are done with loading.
+	if config.JSON.Enabled && !config.JSON.IsDataStream {
+		return nil
+	}
+
+	// If a data stream already exists, we do not attempt to delete or overwrite
+	// it becaues it would delete all backing indices, and the user would lose all
+	// their documents.
+	dataStreamExist, err := l.checkExistsDatastream(templateName)
```

Comment:
Seeing these changes come in after the first iteration really makes me want a test case for this. Could you write an issue to add test coverage for this so we don't lose track of it?

Reply:
I have added integration tests to cover the happy path and possible errors.

Reply:
Thanks! I don't see a test case for this code path where the data stream already exists though, and it seems like it would be easy to add (just run the successful case twice without removing the data stream in between).

The comment here suggests not handling this case is bad enough that we really should have a test for it.